barf/barf.py

```diff
@@ -292,10 +292,17 @@ def recover_cfg(self, ea_start=None, ea_end=None, symbols=None, callback=None, a
         # Set architecture in case it wasn't already set.
         if arch_mode is None:
             arch_mode = self.binary.architecture_mode
-
+        else:
+            if arch_mode == arch.ARCH_ARM:
```

Comment:
arch_mode can only be ARCH_ARM_MODE_ARM and ARCH_ARM_MODE_THUMB. Is this a typo?